src/view/reviewManager.ts

```diff
@@ -1026,13 +1026,83 @@ export class ReviewManager implements vscode.DecorationProvider {
 		await this.validateState();
 	}
 
+	public async publishBranch(): Promise<Branch> {
+		const HEAD = this._repository.state.HEAD;
+		const githubRemotes = this._prManager.getGitHubRemotes();
+		const picks: RemoteQuickPickItem[] = githubRemotes.map(remote => new RemoteQuickPickItem(remote));
+		const selected: RemoteQuickPickItem = await vscode.window.showQuickPick<RemoteQuickPickItem>(picks, {
+			ignoreFocusOut: true,
+			placeHolder: `Pick a remote to publish the branch ${HEAD.name} to:`
+		});
+
+		if (!selected) {
+			return;
+		}
+
+		let upstreamBranchName = await vscode.window.showInputBox({
+			ignoreFocusOut: true,
+			prompt: 'Pick a name for the upstream branch',
+			value: HEAD.name,
+		});
+
+		if (!upstreamBranchName) {
+			return;
+		}
+
+		try {
+			// since we are probably pushing a remote branch with a different name, we use the complete synatx
+			// git push -u origin local_branch:remote_branch
+			await this._repository.push(selected.remote.remoteName, `${HEAD.name}:${upstreamBranchName}`, true);
+		} catch (err) {
+			if (err.gitErrorCode === GitErrorCodes.PushRejected) {
+				let choice = await vscode.window.showWarningMessage(`Can't push refs to remote, try run 'git pull' first to integrate with your change, or choose another remote or branch name.`, {
+					modal: true
+				}, 'Retry');
+
+				if (choice === 'Retry') {
+					return this.publishBranch();
+				}
+
+				return;
+			}
+
+			// we can't handle the error
+			throw err;
+		}
+
+		// we don't want to wait for repository status update
+		let latestBranch = await this._repository.getBranch(HEAD.name);
+		if (!latestBranch || !latestBranch.upstream) {
+			return;
+		}
+
+		return latestBranch;
+	}
+
 	public async createPullRequest(): Promise<void> {
 		const HEAD = this._repository.state.HEAD;
+		let branchName;
```

Comment:
Does this case ever happen? I think it's possible for us to end up in a repo with no GitHub remotes, right?

```suggestion
const potentialTargetRemotes = this._prManager.getGitHubRemotes();
if (!potentialTargetRemotes.length) {
  vscode.window.showWarningMessage(`No GitHub remotes found. Add a remote and try again.`)
}
let branchName;
```

Reply:
thanks, I wish I can use *Code Suggestion* for this one ;)